aten/src/ATen/native/sparse/cuda/SparseBlasImpl.cpp

```diff
@@ -835,9 +835,17 @@ void addmm_out_sparse_csr(
   if (mat1.layout() == kSparseBsr && mat2.layout() == kStrided && result.layout() == kStrided) {
     return block_sparse_mm(mat1, mat2, beta, alpha, result);
   }
+  if (mat1.layout() == kStrided && mat2.layout() == kSparseBsc &&
+      result.layout() == kStrided) {
+    return block_sparse_mm(mat2.t(), mat1.t(), beta, alpha, result.t());
+  }
   if (mat1.is_sparse_csr() && mat2.layout() == kStrided && result.layout() == kStrided) {
     return spmm(mat1, mat2, beta, alpha, result);
   }
+  if (mat1.layout() == kStrided && mat2.layout() == kSparseCsc &&
+      result.layout() == kStrided) {
+    return spmm(mat2.t(), mat1.t(), beta, alpha, result.t());
+  }
```

Comment:
Nit: To facilitate batch inputs, the transpose operations ought to use `.transpose(-2, -1)` instead of `.t()`.

(Furthermore, keeping in mind hybrid sparse support, the transpose method call ought to read `.transpose(nbatchdims, nbatchdims+1)` but perhaps not in this PR as underlying sparse tools may not support hybrid sparse tensors out from the box).

Reply:
I wonder whether `mT` works as well. I am also not sure whether `mH` works, and whether `transpose + conj` is a view on these tensors. I do not remember seeing these things tested, however...

Reply:
yes, `mT` works for all layouts. `mH` works for strided inputs only and is a view.

Reply:
While we do not have a ton of support for hybrid sparse compressed it is starting to pop up. We don't have a mm kernel that could handle them though so we should TORCH_CHECK to exclude that possibility.